libvast/include/vast/system/terminate.hpp

```diff
@@ -31,6 +32,45 @@ struct parallel;
 
 namespace vast::system {
 
+/// Wrapper which extends terminator's lifetime to the response handlers.
+template <class ResponseHandle>
+class termination_result {
```

Comment:
Naming-wise I think it'd be clearer to name this `terminate_result`, because it really is the output of the `terminate` function and not the output of the termination itself. This would be in line with standard library naming as well, e.g., `std::from_chars_result` for `std::from_chars`.

Reply:
Yeah, the name is quite bad. I forgot to change it once i have changed the approach. What do you think about simply terminate_response_handle?

Reply:
I'm fine with either `terminate_result` or `terminate_result_handle`, I think both work fine.